pkg/grapiserver/options.go

```diff
@@ -99,6 +100,14 @@ func WithGrpcServerOptions(opts ...grpc.ServerOption) Option {
 	}
 }
 
+// WithGatewayHTTPServer returns an Option that specifies http.Server configuration to a gateway server.
+// But if you specify Addr or  Handler, they are ignored.
+func WithGatewayHTTPServer(s http.Server) Option {
+	return func(c *Config) {
+		c.GatewayHTTPServer = s
```

Comment:
**[govet]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
assignment copies lock value to c.GatewayHTTPServer: net/http.Server contains sync.Once contains sync.Mutex